prow/githubeventserver/options.go

```diff
@@ -49,6 +49,11 @@ func (o *Options) Validate() error {
 	if !strings.HasPrefix(o.endpoint, "/") {
 		return fmt.Errorf("endpoint %s is not a valid url path", o.endpoint)
 	}
+
+	o.hmacTokenGenerator = func() []byte { return []byte("") }
```

Comment:
This is defaulting, not validation. `Validate` will now unconditinally overwrite all existing options. This doesn't make any sense.

Reply:
This is just an initialization, to avoid those fields to be nil.

Reply:
No. You set options, then you validate them. With what you do here, doing that will just overwrite everything a user set unconditionally. This doesn't make any sense.

Reply:
In the validation point, the user won't set anything. The option flags are setting `endpoint` and `port` only. The rest stays nil. I am fine with that and rely on the user to set them later, so I can remove this change. WDYT?

Reply:
Just make the fields pointers, check if they are nil, only default them if they are nil, rename the func to `DefaultAndValidate`